src/unix/thread.c

```diff
@@ -646,13 +646,16 @@ void uv_cond_wait(uv_cond_t* cond, uv_mutex_t* mutex) {
 int uv_cond_timedwait(uv_cond_t* cond, uv_mutex_t* mutex, uint64_t timeout) {
   int r;
   struct timespec ts;
+  struct timeval tv;
 
 #if defined(__APPLE__) && defined(__MACH__)
   ts.tv_sec = timeout / NANOSEC;
   ts.tv_nsec = timeout % NANOSEC;
   r = pthread_cond_timedwait_relative_np(cond, mutex, &ts);
 #else
-  timeout += uv__hrtime(UV_CLOCK_PRECISE);
+  if (gettimeofday(&tv, NULL))
+    abort();
+  timeout += (tv.tv_sec * NANOSEC) + (tv.tv_usec * 1e3);
```

Comment:
Can you move this to the `#else` branch below?  It's incorrect for the android path.  The `uv__hrtime()` call should be moved there.